src/main/java/org/terasology/snakeTournament/systems/BlockTrailAuthoritySystem.java

```diff
@@ -40,13 +28,13 @@
     @ReceiveEvent
     public void onCharacterMoved(MovedEvent event, EntityRef character) {
         // calculate our previous position by subtracting the delta
-        Vector3f previousPosition = new Vector3f(event.getPosition());
-        previousPosition.sub(event.getDelta());
+        Vector3f currentPosition = new Vector3f(event.getPosition());
+        Vector3f previousPosition = currentPosition.sub(event.getDelta(), new Vector3f());
 
         // find out the current and previous block positions
         // offset by 0.5 to account for the fact that blocks are center aligned on the whole numbers
-        Vector3i previousBlockPosition = new Vector3i(previousPosition, 0.5f);
-        Vector3i currentBlockPosition = new Vector3i(event.getPosition(), 0.5f);
+        Vector3i currentBlockPosition = new Vector3i(currentPosition.add(new Vector3f(0.5f, 0.5f, 0.5f)), RoundingMode.FLOOR);
```

Comment:
Let's wait a moment and use `Blocks.toBlockPos(currentPosition)` here (https://github.com/MovingBlocks/Terasology/pull/4516)